From 1cf2dce616585334e6c2addc8b577b4f89f65b7d Mon Sep 17 00:00:00 2001
From: Kondapally Kalyan <kalyan.kondapally@intel.com>
Date: Sat, 7 Dec 2013 12:01:02 +0200
Subject: [PATCH 1/3] Ash shell changes.

---
 ash/shell.cc |   47 ++++++++++++++++++++++++++---------------------
 ash/shell.h  |    6 ++----
 2 files changed, 28 insertions(+), 25 deletions(-)

diff --git a/ash/shell.cc b/ash/shell.cc
index 738eeac..f940977 100644
--- a/ash/shell.cc
+++ b/ash/shell.cc
@@ -117,16 +117,19 @@
 
 #if defined(OS_CHROMEOS)
 #if defined(USE_X11)
+#include "chromeos/display/output_configurator_x11.h"
+#elif defined(USE_OZONE)
+#include "ui/ozone/ozone_platform.h"
+#endif  // defined(USE_X11)
+
 #include "ash/ash_constants.h"
 #include "ash/display/display_change_observer_chromeos.h"
 #include "ash/display/display_error_observer_chromeos.h"
 #include "ash/display/output_configurator_animation.h"
-#include "base/message_loop/message_pump_x11.h"
 #include "base/sys_info.h"
 #include "chromeos/display/output_configurator.h"
 #include "content/public/browser/gpu_data_manager.h"
 #include "gpu/config/gpu_feature_type.h"
-#endif  // defined(USE_X11)
 #include "ash/system/chromeos/brightness/brightness_controller_chromeos.h"
 #include "ash/system/chromeos/power/power_event_observer.h"
 #include "ash/system/chromeos/power/power_status.h"
@@ -160,6 +163,19 @@ class AshVisibilityController : public views::corewm::VisibilityController {
   DISALLOW_COPY_AND_ASSIGN(AshVisibilityController);
 };
 
+#if defined(OS_CHROMEOS)
+chromeos::OutputConfigurator* CreateOuputConfigurator() {
+#if defined(USE_OZONE)
+  ui::OzonePlatform::Initialize();
+  return chromeos::OutputConfiguratorOzone::GetInstance();
+#elif defined(USE_X11)
+  return new chromeos::OutputConfiguratorX11();
+#endif
+  NOTREACHED();
+  return 0;
+}
+#endif
+
 }  // namespace
 
 // static
@@ -554,9 +570,9 @@ Shell::Shell(ShellDelegate* delegate)
       delegate_(delegate),
       window_positioner_(new WindowPositioner),
       activation_client_(NULL),
-#if defined(OS_CHROMEOS) && defined(USE_X11)
-      output_configurator_(new chromeos::OutputConfigurator()),
-#endif  // defined(OS_CHROMEOS) && defined(USE_X11)
+#if defined(OS_CHROMEOS)
+      output_configurator_(CreateOuputConfigurator()),
+#endif  // defined(OS_CHROMEOS)
       native_cursor_manager_(new AshNativeCursorManager),
       cursor_manager_(scoped_ptr<views::corewm::NativeCursorManager>(
           native_cursor_manager_)),
@@ -570,22 +586,14 @@ Shell::Shell(ShellDelegate* delegate)
   if (!gfx::Screen::GetScreenByType(gfx::SCREEN_TYPE_NATIVE))
     gfx::Screen::SetScreenInstance(gfx::SCREEN_TYPE_NATIVE, screen_);
   display_controller_.reset(new DisplayController);
-#if defined(OS_CHROMEOS) && defined(USE_X11)
+#if defined(OS_CHROMEOS)
   bool is_panel_fitting_disabled =
       content::GpuDataManager::GetInstance()->IsFeatureBlacklisted(
           gpu::GPU_FEATURE_TYPE_PANEL_FITTING);
 
   output_configurator_->Init(!is_panel_fitting_disabled);
   periodic_metrics_recorder_.reset(new PeriodicMetricsRecorder);
-
-  base::MessagePumpX11::Current()->AddDispatcherForRootWindow(
-      output_configurator());
-  // We can't do this with a root window listener because XI_HierarchyChanged
-  // messages don't have a target window.
-  base::MessagePumpX11::Current()->AddObserver(output_configurator());
-#endif  // defined(OS_CHROMEOS)
-
-#if defined(OS_CHROMEOS)
+  output_configurator_->StartProcessingEvents();
   internal::PowerStatus::Initialize();
 #endif
 }
@@ -699,17 +707,14 @@ Shell::~Shell() {
   new_window_delegate_.reset();
   media_delegate_.reset();
 
-#if defined(OS_CHROMEOS) && defined(USE_X11)
+#if defined(OS_CHROMEOS)
    if (display_change_observer_)
     output_configurator_->RemoveObserver(display_change_observer_.get());
   if (output_configurator_animation_)
     output_configurator_->RemoveObserver(output_configurator_animation_.get());
   if (display_error_observer_)
     output_configurator_->RemoveObserver(display_error_observer_.get());
-  base::MessagePumpX11::Current()->RemoveDispatcherForRootWindow(
-      output_configurator());
-  base::MessagePumpX11::Current()->RemoveObserver(output_configurator());
-  display_change_observer_.reset();
+  output_configurator_->StopProcessingEvents();
 #endif  // defined(OS_CHROMEOS)
 
 #if defined(OS_CHROMEOS)
@@ -729,7 +734,7 @@ void Shell::Init() {
         internal::DisplayManager::VIRTUAL_KEYBOARD);
   }
   bool display_initialized = display_manager_->InitFromCommandLine();
-#if defined(OS_CHROMEOS) && defined(USE_X11)
+#if defined(OS_CHROMEOS)
   output_configurator_animation_.reset(
       new internal::OutputConfiguratorAnimation());
   output_configurator_->AddObserver(output_configurator_animation_.get());
diff --git a/ash/shell.h b/ash/shell.h
index b387d19..4d60023 100644
--- a/ash/shell.h
+++ b/ash/shell.h
@@ -481,7 +481,7 @@ class ASH_EXPORT Shell
   // Starts the animation that occurs on first login.
   void DoInitialWorkspaceAnimation();
 
-#if defined(OS_CHROMEOS) && defined(USE_X11)
+#if defined(OS_CHROMEOS)
   // TODO(oshima): Move these objects to DisplayController.
   chromeos::OutputConfigurator* output_configurator() {
     return output_configurator_.get();
@@ -492,7 +492,7 @@ class ASH_EXPORT Shell
   internal::DisplayErrorObserver* display_error_observer() {
     return display_error_observer_.get();
   }
-#endif  // defined(OS_CHROMEOS) && defined(USE_X11)
+#endif  // defined(OS_CHROMEOS)
 
   internal::ResolutionNotificationController*
       resolution_notification_controller() {
@@ -672,7 +672,6 @@ class ASH_EXPORT Shell
   scoped_ptr<internal::UserActivityNotifier> user_activity_notifier_;
   scoped_ptr<internal::VideoActivityNotifier> video_activity_notifier_;
   scoped_ptr<StickyKeys> sticky_keys_;
-#if defined(USE_X11)
   // Controls video output device state.
   scoped_ptr<chromeos::OutputConfigurator> output_configurator_;
   scoped_ptr<internal::OutputConfiguratorAnimation>
@@ -681,7 +680,6 @@ class ASH_EXPORT Shell
 
   // Listens for output changes and updates the display manager.
   scoped_ptr<internal::DisplayChangeObserver> display_change_observer_;
-#endif  // defined(USE_X11)
 #endif  // defined(OS_CHROMEOS)
 
   scoped_ptr<internal::ResolutionNotificationController>
-- 
1.7.9.5

