From c84ba6e93f919315772d936ffd45744a720c7461 Mon Sep 17 00:00:00 2001
From: Kondapally Kalyan <kalyan.kondapally@intel.com>
Date: Sat, 7 Dec 2013 14:15:29 +0200
Subject: [PATCH 2/3] Include DBUS with Ozone.

---
 ui/base/ime/ime.gypi |    2 --
 1 file changed, 2 deletions(-)

diff --git a/ui/base/ime/ime.gypi b/ui/base/ime/ime.gypi
index ae8198b..a88693b 100644
--- a/ui/base/ime/ime.gypi
+++ b/ui/base/ime/ime.gypi
@@ -84,8 +84,6 @@
       'sources!': [
         'character_composer.cc',
         'character_composer.h',
-	'chromeos/ibus_bridge.cc',
-        'chromeos/ibus_bridge.h',
 	'chromeos/mock_ime_candidate_window_handler.cc',
         'chromeos/mock_ime_candidate_window_handler.h',
         'chromeos/mock_ime_engine_handler.cc',
-- 
1.7.9.5

